test/widgets/test_clipboard.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2021 elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import textwrap
+
+import pytest
+
+import libqtile.config
+import libqtile.widget.bluetooth
+from libqtile.bar import Bar
+from libqtile.config import Screen
+from test.helpers import Retry
+
+
+@Retry(ignore_exceptions=(AssertionError, ))
+def clipboard_cleared(widget):
+    assert widget.info()["text"] == ""
+
+
+@pytest.fixture
+def clipboard_manager(request, minimal_conf_noscreen, manager_nospawn):
+    widget = libqtile.widget.Clipboard(**getattr(request, "param", dict()))
+    config = minimal_conf_noscreen
+    config.screens = [
+        Screen(
+            top=Bar([widget], 10)
+        )
+    ]
+    manager_nospawn.start(config)
+
+    if manager_nospawn.backend.name != "x11":
+        pytest.skip("Test only available on X11.")
```

Comment:
I've assumed this widget should be X11 only but the tests would probably still work on Wayland as I'm just mocking the events.

Reply:
Currently yeah